### PR TITLE
headache: 1.06 -> 1.07

### PR DESCRIPTION
**Pull request by Niols**
###### Description of changes

- Bump from 1.06 to 1.07:
  - Release: https://github.com/Frama-C/headache/releases/tag/v1.07
  - Changes: https://github.com/frama-c/headache/compare/v1.06...v1.07
  - **Important**: Headache v1.07 depends on Camomile v2. This PR builds on top of https://github.com/NixOS/nixpkgs/pull/242103 and should therefore only be merged afterwards.

- Add `passthru.updateScript`:
  - Allows to use `nix-shell maintainers/scripts/update.nix --argstr package headache`
  - Should hook into general nixpkgs autoupdate process.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Niols** [comment]
Done.

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 242104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>headache</li>
  </ul>
</details>

File: pkgs/development/tools/headache/default.nix [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
   src = fetchFromGitHub {
     owner = "frama-c";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-BA7u09MKYMyspFX8AcAkDVA6UUG5DKAdbIDdt+b3Fc4=";
+    sha256 = "sha256-RL80ggcJSJFu2UTECUNP6KufRhR8ZnG7sQeYzhrw37g=";
   };
 
   propagatedBuildInputs = [
     (camomile.override { version = "1.0.2"; })
```
Comment (ulrikstrid) [inline pkgs/development/tools/headache/default.nix:17]:
```suggestion
    camomile
```
Headache has been updated to camomile 2.0 according to release notes, and it doesn't build with current setup

**Review by ulrikstrid — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/headache/default.nix (hunk 1 above)
Comment (Niols) [reply pkgs/development/tools/headache/default.nix:17]: Indeed! I think it's what made me start #242103 months ago!

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 242104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>headache</li>
  </ul>
</details>

**Review by ulrikstrid — APPROVED** [review]
Looks good now, I guess we would want the commits squashed? @vbgl

**Comment by Niols** [comment]
I think it makes sense to keep two separate commits, one updating Headache and one adding the `updateScript`, but I could squash your suggestion into the former.

**Comment by ulrikstrid** [comment]
Your suggestion sounds good @Niols, if you do that I can merge

**Comment by Niols** [comment]
Done!

**Pull request merged**